cloudbridge/providers/aws/services.py

```diff
@@ -782,7 +782,16 @@ def create(self, label, image, vm_type, subnet,
             # pylint:disable=protected-access
             inst[0]._wait_till_exists()
             # Tag the instance w/ the name
-            inst[0].label = label
+            try:
+                inst[0].label = label
```

Comment:
The tests keep keep failing [here](https://travis-ci.com/github/CloudVE/cloudbridge/jobs/441076339#L1071). Should this be wrapped into a method with retry as well? Same for the security group label and description (https://github.com/CloudVE/cloudbridge/blob/144910ff7787d848a5cbcf9f22494a1984cb36f4/cloudbridge/providers/aws/services.py#L194)?

Reply:
I should have used `inst[0].delete()` instead of `inst.delete()`. Should be fixed now.

Reply:
That fixed the instance labeling failures but setting the security group labels is facing the same issue now so should it be wrapped? 
More generally, as you keep saying, it really seems the `wait_till_ready` is not functioning as is should from the AWS side. This [failure to create an image](https://travis-ci.com/github/CloudVE/cloudbridge/jobs/441127560#L993) should not have happened tracing the calls... I guess more retries will need to be added for the time being.

Reply:
Have added a retry to every label setter. I think we need to think about a more comprehensive plan here. Maybe every operation should atomically succeed or fail. If it fails, it should presumably undo what it did, such as deleting a volume it just created.